File: python/gate 4/level_five.py

def get_sorted_prime_numbers(numbers):
    prime_numbers = []
    for number in numbers:
        if number > 1:
            is_prime = True
            for count in range(2, number):
                if number % count == 0:
                    is_prime = False
            if is_prime:
                prime_numbers.append(number)
    
    
    for count in range(len(prime_numbers)):
        for counter in range(count + 1, len(prime_numbers)):
            if prime_numbers[count] > prime_numbers[counter]:
                prime_numbers[count], prime_numbers[counter] = prime_numbers[counter], prime_numbers[count]
    
    return prime_numbers



def replace_negatives_with_zero(numbers):
    result = []
    for number in numbers:
        if number < 0:
            result.append(0)
        else:
            result.append(number)
    return result



def move_zeros_to_end(numbers):
    no_zeros = []
    zero_count = 0
    for number in numbers:
        if number != 0:
            no_zeros.append(number)
        else:
            zero_count += 1
    for count in range(zero_count):
        no_zeros.append(0)
    return no_zeros



def get_duplicate_elements(numbers):
    duplicates = []
    for count in range(len(numbers)):
        for counter in range(count + 1, len(numbers)):
            if numbers[count] == numbers[counter]:
                if numbers[count] not in duplicates:
                    duplicates.append(numbers[count])
    return duplicates

File: java/gate 4/LevelFive.java
public class LevelFive {

    public static int[] getSortedPrimeNumbers(int[] numbers) {
        int primeCount = 0;
        for (int number : numbers) {
            if (number > 1) {
                boolean isPrime = true;
                for (int divisor = 2; divisor < number; divisor++) {
                    if (number % divisor == 0) {
                        isPrime = false;
                    }
                }
                if (isPrime) {
                    primeCount++;
                }
            }
        }
        int[] primeNumbers = new int[primeCount];
        int primeIndex = 0;
        for (int number : numbers) {
            if (number > 1) {
                boolean isPrime = true;
                for (int divisor = 2; divisor < number; divisor++) {
                    if (number % divisor == 0) {
                        isPrime = false;
                    }
                }
                if (isPrime) {
                    primeNumbers[primeIndex++] = number;
                }
            }
        }
        for (int outerIndex = 0; outerIndex < primeNumbers.length; outerIndex++) {
            for (int innerIndex = outerIndex + 1; innerIndex < primeNumbers.length; innerIndex++) {
                if (primeNumbers[outerIndex] > primeNumbers[innerIndex]) {
                    int temp = primeNumbers[outerIndex];
                    primeNumbers[outerIndex] = primeNumbers[innerIndex];
                    primeNumbers[innerIndex] = temp;
                }
            }
        }
        return primeNumbers;
    }

    public static int[] replaceNegativesWithZero(int[] numbers) {
        int[] result = new int[numbers.length];
        for (int index = 0; index < numbers.length; index++) {
            if (numbers[index] < 0) {
                result[index] = 0;
            } else {
                result[index] = numbers[index];
            }
        }
        return result;
    }

    public static int[] moveZerosToEnd(int[] numbers) {
        int zeroCount = 0;
        int nonZeroIndex = 0;
        for (int number : numbers) {
            if (number == 0) {
                zeroCount++;
            }
        }
        int[] result = new int[numbers.length];
        for (int number : numbers) {
            if (number != 0) {
                result[nonZeroIndex++] = number;
            }
        }
        for (int index = numbers.length - zeroCount; index < numbers.length; index++) {
            result[index] = 0;
        }
        return result;
    }

    public static int[] getDuplicateElements(int[] numbers) {
        int duplicateCount = 0;
        for (int outerIndex = 0; outerIndex < numbers.length; outerIndex++) {
            for (int innerIndex = outerIndex + 1; innerIndex < numbers.length; innerIndex++) {
                if (numbers[outerIndex] == numbers[innerIndex]) {
                    boolean alreadyAdded = false;
                    for (int checkIndex = 0; checkIndex < duplicateCount; checkIndex++) {
                        if (numbers[outerIndex] == numbers[checkIndex]) {
                            alreadyAdded = true;
                        }
                    }
                    if (!alreadyAdded) {
                        duplicateCount++;
                    }
                    break;
                }
            }
        }
        int[] duplicates = new int[duplicateCount];
        int duplicateIndex = 0;
        for (int outerIndex = 0; outerIndex < numbers.length; outerIndex++) {
            for (int innerIndex = outerIndex + 1; innerIndex < numbers.length; innerIndex++) {
                if (numbers[outerIndex] == numbers[innerIndex]) {
                    boolean alreadyAdded = false;
                    for (int checkIndex = 0; checkIndex < duplicateIndex; checkIndex++) {
                        if (duplicates[checkIndex] == numbers[outerIndex]) {
                            alreadyAdded = true;
                        }
                    }
                    if (!alreadyAdded) {
                        duplicates[duplicateIndex++] = numbers[outerIndex];
                    }
                    break;
                }
            }
        }
        return duplicates;
    }
}
